src/components/theme-switch/index.tsx

```diff
@@ -0,0 +1,24 @@
+import { useState, useEffect } from 'react'
+import { useTheme } from 'next-themes'
+
+export default function ThemeSwitch() {
+	const [mounted, setMounted] = useState(false)
+	const { theme, setTheme } = useTheme()
+
+	// useEffect only runs on the client, so now we can safely show the UI
+	useEffect(() => {
+		setMounted(true)
+	}, [])
+
+	if (!mounted) {
+		return null
+	}
+
+	return (
+		<select value={theme} onChange={(e) => setTheme(e.target.value)}>
```

Comment:
[Copied this from their docs](https://github.com/pacocoursey/next-themes#avoid-hydration-mismatch), the system default and having that as an item in the dropdown select is slick.